core/src/loader.rs

```diff
@@ -1089,20 +1092,61 @@ impl<'gc> Loader<'gc> {
 
                         Avm2::dispatch_event(&mut activation.context, progress_evt, target);
 
+                        let http_status_evt = activation
+                            .avm2()
+                            .classes()
+                            .httpstatusevent
+                            .construct(
+                                &mut activation,
+                                &[
+                                    "httpStatus".into(),
+                                    false.into(),
+                                    false.into(),
+                                    response.status.into(),
+                                    response.redirected.into(),
+                                ],
+                            )
+                            .map_err(|e| Error::Avm2Error(e.to_string()))?;
+
+                        Avm2::dispatch_event(&mut activation.context, http_status_evt, target);
+
                         let complete_evt = Avm2EventObject::bare_default_event(
                             &mut activation.context,
                             "complete",
                         );
                         Avm2::dispatch_event(uc, complete_evt, target);
                     }
-                    Err(_err) => {
+                    Err(err) => {
                         // Testing with Flash shoes that the 'data' property is cleared
                         // when an error occurs
 
                         set_data(Vec::new(), &mut activation, target, data_format);
 
-                        // FIXME - Match the exact error message generated by Flash
+                        let status_code = if let Error::HTTPNotOK(_, status_code) = err {
+                            status_code
+                        } else {
+                            0
+                        };
+
+                        let http_status_evt = activation
+                            .avm2()
+                            .classes()
+                            .httpstatusevent
+                            .construct(
+                                &mut activation,
+                                &[
+                                    "httpStatus".into(),
+                                    false.into(),
+                                    false.into(),
+                                    status_code.into(),
+                                    false.into(),
```

Comment:
Shouldn't this be set based on whether the failing request was redirected or not?